tests/testthat/test-implicit_integer_linter.R

```diff
@@ -85,22 +85,31 @@ test_that("linter returns the correct linting", {
 })
 
 skip_if_not_installed("tibble")
-patrick::with_parameters_test_that(
-  "numbers in a:b input are optionally not linted",
-  expect_lint(
-    paste0(left, ":", right),
-    if (n_lints > 0L) rep(list(rex::rex("Use 1L or 1.0")), n_lints),
-    implicit_integer_linter(allow_colon = allow_colon)
-  ),
-  .cases = tibble::tribble(
-    ~left,  ~right, ~n_lints, ~allow_colon, ~.test_name,
-    "1",    "1",    2L,       FALSE,        "1:1, !allow_colon",
-    "1",    "1",    0L,       TRUE,         "1:1, allow_colon",
-    "1",    "1L",   1L,       FALSE,        "1:1L, !allow_colon",
-    "1",    "1L",   0L,       TRUE,         "1:1L, allow_colon",
-    "1L",   "1",    1L,       FALSE,        "1L:1, !allow_colon",
-    "1L",   "1",    0L,       TRUE,         "1L:1, allow_colon",
-    "1L",   "1L",   0L,       FALSE,        "1L:1L, !allow_colon",
-    "1L",   "1L",   0L,       TRUE,         "1L:1L, allow_colon"
+local({
+  .cases <- expand.grid(
+    left = c("1", "-1", "1L", "-1L"),
+    right = c("1", "-1", "1L", "-1L"),
+    allow_colon = c(FALSE, TRUE),
+    stringsAsFactors = FALSE
```

Comment:
Is this necessary on our supported versions?

Reply:
Yes, `expand.grid()` still uses that default :(

https://github.com/r-devel/r-svn/blob/69e84f0ed657638d152918392f33e875ee400d58/src/library/base/R/expand.grid.R#L19